src/libs/DebugUtils.ts

```diff
@@ -0,0 +1,587 @@
+/* eslint-disable max-classes-per-file */
+import {isMatch} from 'date-fns';
+import isValid from 'date-fns/isValid';
+import type {OnyxEntry} from 'react-native-onyx';
+import CONST from '@src/CONST';
+import type {Report, ReportAction} from '@src/types/onyx';
+
+class NumberError extends SyntaxError {
+    constructor() {
+        super('debug.invalidValue', {cause: {expectedValues: 'number | undefined | ""'}});
+    }
+}
+
+class ArrayError extends SyntaxError {
+    constructor(arrayType: string | Record<string, unknown>) {
+        super('debug.invalidValue', {
+            cause: {
+                expectedValues: `[${typeof arrayType === 'object' ? stringifyJSON(arrayType) : arrayType}]`,
+            },
+        });
+    }
+}
+
+class ObjectError extends SyntaxError {
+    constructor(type: Record<string, unknown>) {
+        super('debug.invalidValue', {
+            cause: {
+                expectedValues: `${stringifyJSON(type)} | undefined | ''`,
+            },
+        });
+    }
+}
+
+type ObjectType = Record<
+    string,
+    | 'string'
+    | 'number'
+    | 'object'
+    | 'array'
+    | 'boolean' // Constant enum
```

Comment:
Is this comment needed?
```suggestion
    | 'boolean'
```

Reply:
It's not, I'm removing it thanks! 😄